dali/test/python/test_operator_reduce.py

```diff
@@ -124,31 +151,54 @@ def compare(dali_res, np_res):
         assert np.array_equal(dali_sample, np_sample)
 
 
-def run_reduce(keep_dims, reduce_fns, batch_gen, data_type):
-    batch_fn = batch_gen(data_type)
+def run_reduce(keep_dims, reduce_fns, batch_gen, input_type, output_type = None):
+    batch_fn = batch_gen(input_type)
     dali_reduce_fn, numpy_reduce_fn = reduce_fns
 
     for axes in batch_fn.valid_axes():
         dali_res_cpu, dali_res_gpu = run_dali(
-            dali_reduce_fn, batch_fn, keep_dims = keep_dims, axes = axes)
+            dali_reduce_fn, batch_fn, keep_dims = keep_dims, axes = axes, output_type = output_type)
 
         batch_fn.reset()
 
         np_res = run_numpy(
-            numpy_reduce_fn, batch_fn, keep_dims = keep_dims, axes = axes)
+            numpy_reduce_fn, batch_fn, keep_dims = keep_dims, axes = axes, output_type = output_type)
         
         for iteration in range(batch_fn.num_iter()):
             compare(dali_res_cpu[iteration], np_res[iteration])
             compare(dali_res_gpu[iteration], np_res[iteration])
 
 
 def test_reduce():
-    reductions = [(fn.min, np.min), (fn.max, np.max)]
+    reductions = [(fn.sum, np.sum), (fn.min, np.min), (fn.max, np.max)]
+
     batch_gens = [Batch1D, Batch2D, Batch3D]
-    types = [np.uint8, np.int8, np.uint16, np.int16, np.uint32, np.int32, np.uint64, np.int64, np.float32]
+    types = [
+        np.uint8, np.int8, np.uint16, np.int16, np.uint32, np.int32, np.uint64, np.int64, np.float32]
 
-    for keep_dims in [True, False]:
+    for keep_dims in [False, True]:
         for reduce_fns in reductions:
             for batch_gen in batch_gens:
                 for type_id in types:
                     yield run_reduce, keep_dims, reduce_fns, batch_gen, type_id
+
+
+def test_reduce_with_output_type():
+    reductions = [(fn.sum, np.sum)]
+
+    batch_gens = [Batch3DOverflow]
+    types = [
+        (np.uint8, [np.uint64, np.float32]),
+        (np.int8, [np.int64, np.float32]),
+        (np.uint16, [np.uint64, np.float32]),
+        (np.int16, [np.int64, np.float32]),
+        (np.uint32, [np.uint64, np.float32]),
+        (np.int32, [np.int32, np.int64, np.float32])]
+
+    for keep_dims in [False, True]:
```

Comment:
How long does this test take. Maybe we should split it to smaller and bigger flavor?

Reply:
This whole file as it is in this PR takes around 30 seconds to run, so I don't think it's a problem.